src/modules/launcher/Wox.Plugin/Common/ShellLocalization.cs

```diff
@@ -39,11 +41,14 @@ public string GetLocalizedName(string path)
 
             shellItem.GetDisplayName(SIGDN.NORMALDISPLAY, out string filename);
 
-            if (!_localizationCache.ContainsKey(path.ToLowerInvariant()))
+            lock (_cacheLock)
```

Comment:
Shift to a [ConcurrentDictionary](https://learn.microsoft.com/dotnet/api/system.collections.concurrent.concurrentdictionary-2) instead of the lock may be an improvement?